### PR TITLE
Load secrets via ESO for eks build clusters

### PR DESCRIPTION
**Pull request by upodroid**
/cc @xmudrii @ameukam 

I wonder what the secret contents look like. You might have to do something like this if they are base64 encoded and stored as json:

```
apiVersion: external-secrets.io/v1beta1
kind: ExternalSecret
metadata:
  name: gcs-upload
  namespace: test-pods
spec:
  refreshInterval: 10m
  secretStoreRef:
    kind: ClusterSecretStore
    name: knative-tests
  target:
    creationPolicy: Owner
    template:
      type: Opaque
      engineVersion: v2
      data:
        service-account.json: "{{ .test_account | toString | b64dec}}" # Used by gcsupload decorator
  data:
  - secretKey: test_account
    remoteRef:
      key: prow-google-service-accounts
      property: test_account
```

### CODE REVIEW

**Comment by upodroid** [comment]
@ameukam Did you deploy the infra changes from #4930 ?

File: infra/aws/terraform/prow-build-cluster/resources/test-pods/externalsecrets.yaml [hunk 1]
```diff
@@ -4,78 +4,114 @@
...
-#     name: aws-ssh-private                                                        # The key to write to in the Kubernetes Secret
-#     version: latest                                                              # The version of the GSM Secret
-#     property: aws-ssh-private
-#   - key: gke_k8s-prow-builds_us-central1-f_prow__test-pods__aws-ssh-key-secret   # The name of the GSM Secret
-#     name: aws-ssh-public                                                         # The key to write to in the Kubernetes Secret
-#     version: latest                                                              # The version of the GSM Secret
-#     property: aws-ssh-public
+---
+apiVersion: external-secrets.io/v1beta1
+kind: ClusterSecretStore
```
Comment (ameukam) [inline infra/aws/terraform/prow-build-cluster/resources/test-pods/externalsecrets.yaml:9]: We only want to sync secrets in a specific namespace. we should scope this to the namespace.

**Comment by xmudrii** [comment]
@upodroid I noticed that sync time for ExternalSecrets is 1 hour. That seems too long, AFAIK, it's 10 seconds for the Prow control plane cluster. Should we reduce that to like 10 minutes at least?

File: infra/aws/terraform/prow-build-cluster/resources/test-pods/externalsecrets.yaml [hunk 2]
```diff
@@ -4,78 +4,114 @@
...
+  data:
+  - secretKey: service-account.json # The key to write to in the Kubernetes Secret
+    remoteRef:
+      key: prow-build-service-account # The name of the GSM Secret
+      version: latest                 # The version of the GSM Secret    
+---
+apiVersion: external-secrets.io/v1beta1
+kind: ExternalSecret
+metadata:
+  name: ssh-key-secret
```
Comment (ameukam) [inline infra/aws/terraform/prow-build-cluster/resources/test-pods/externalsecrets.yaml:38]: We don't need this SSH key. This is needed for e2e tests on GCP. we may need to create a new key pair on EC2.

**Comment by upodroid** [comment]
Yes

**Pull request closed without merging**